services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/RuleServiceImpl.java

```diff
@@ -73,6 +79,120 @@ public ResponseEntity<Object> saveRule(RuleDetails ruleDetails, int userId, int
 		final Map<String, Object> response = new HashMap<>();
 		log.info("Invoking saveRule  API {}", ruleDetails.toString());
 		try {
+			String targetFeedName = null;
+			String targetFileName = null;
+			String targetBucketName = null;
+			String targetFilePattern = null;
+			if(ruleDetails.isSourceAndTarget()) {
+				DataSourceDetails targetDetails = ruleDetails.getTargetData();
+				if(targetDetails == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target details can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+                targetFilePattern = targetDetails.getFilePattern();
+				if(targetFilePattern == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target file pattern can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+				String filePath = targetDetails.getFilePath();
+				if(filePath == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target file path can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+				targetBucketName = targetDetails.getBucketName();
+				String targetFilePath = "s3://" + targetBucketName + "/" + filePath;
+
+				// Split strings
+				String[] targetFilePatternList = targetFilePattern.split("/");
+				String[] targetFilePathList = targetFilePath.split("/");
+				// Check if both arrays have the same size
+				if (targetFilePatternList.length != targetFilePathList.length) {
+					log.info(FILE_NOT_ALIGN);
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", FILE_NOT_ALIGN);
+					return ResponseEntity.ok().body(response);
+				}
+				int targetFeedNameIndex = PatternUtils.getIndex(targetFilePatternList,FEED_NAME);
+				if(targetFeedNameIndex == -1){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", INDEX_NOT_FOUND);
+					return ResponseEntity.ok().body(response);
+				}
+				int targetDateIndex = PatternUtils.getIndex(targetFilePatternList,DDMMYYYY);
+				if(targetDateIndex == -1){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", INDEX_NOT_FOUND);
+					return ResponseEntity.ok().body(response);
+				}
+				int targetFileNameIndex = PatternUtils.getIndex(targetFilePatternList,FILENAME);
+				if(targetFileNameIndex == -1){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", INDEX_NOT_FOUND);
+					return ResponseEntity.ok().body(response);
+				}
+				targetFeedName = PatternUtils.getValueAtIndex(targetFilePathList, targetFeedNameIndex);
+				String targetDate = PatternUtils.getValueAtIndex(targetFilePathList, targetDateIndex);
+				targetFileName = PatternUtils.getValueAtIndex(targetFilePathList, targetFileNameIndex);
```

Comment:
line 140 is having null.. whats the use of getting this targetIndex , if sourceDetails is null?